components/org.wso2.micro.integrator.initializer/src/main/java/org/wso2/micro/integrator/initializer/deployment/AppDeployerServiceComponent.java

```diff
@@ -210,4 +225,40 @@ public void registerEventSinkDeployer(ArtifactDeploymentManager artifactDeployme
             log.error("Error instantiating EventSinkDeployer class", e);
         }
     }
+
+    /**
+     * Initialize and add the CappDeployer to the Deployment Engine.
+     */
+    private void addCAppDeployer() {
+        try {
+            String appsRepo = "carbonapps";
+            // Initialize CApp deployer here
+            Class deployerClass = Class.
+                    forName("org.wso2.micro.integrator.initializer.deployment.application.deployer.CappDeployer");
+            Deployer deployer = (Deployer) deployerClass.newInstance();
+
+            //Add the deployer to deployment engine
+            DeploymentEngine deploymentEngine =
+                    (DeploymentEngine) configCtx.getAxisConfiguration().getConfigurator();
+            deploymentEngine.addDeployer(deployer, appsRepo, "car");
+        } catch (ClassNotFoundException e) {
+            log.error("Can not find class CappDeployer", e);
+        } catch (IllegalAccessException | InstantiationException e) {
+            log.error("Error while instantiating CappDeployer class", e);
+        }
+    }
+
+    /**
+     * Invoke all registered deployers.
+     */
+    private void invokeRegisteredDeployers() {
+        try {
+            AxisConfigurator axisConfigurator = configCtx.getAxisConfiguration().getConfigurator();
+            if (axisConfigurator instanceof CarbonAxisConfigurator) {
+                ((CarbonAxisConfigurator) axisConfigurator).deployServices();
+            }
+        } catch (Exception e) {
```

Comment:
Let's not catch a generic exception here. Also, I couldn't find any exception thrown from the methods inside the try block.  Shall we please check?

Reply:
Adding this exception handling part was done in EI as a separate commit in EI here https://github.com/wso2/carbon-kernel/commit/9a1c9aabcecfd1bdf76a41c5fdbe97ef82bfbe03 

However, considering the artifact deployment as a critical task, let's not handle the exception in the catch block.